test/integration/e2e/end_to_end.go

```diff
@@ -227,3 +234,189 @@ func createChannel(t *testing.T, sdk *fabsdk.FabricSDK, resMgmtClient *resmgmt.C
 	require.Nil(t, err, "error should be nil")
 	require.NotEmpty(t, txID, "transaction ID should be populated")
 }
+
+// createCCLifecycle package cc, install cc, get installed cc package, query installed cc
+// approve cc, query approve cc, check commit readiness, commit cc, query committed cc
+func createCCLifecycle(t *testing.T, orgResMgmt *resmgmt.Client, sdk *fabsdk.FabricSDK) {
+	// Package cc
+	label, ccPkg := packageCC(t)
+	packageID := lcpackager.ComputePackageID(label, ccPkg)
+
+	// Install cc
+	installCC(t, label, ccPkg, orgResMgmt)
+
+	// Get installed cc package
+	getInstalledCCPackage(t, packageID, ccPkg, orgResMgmt)
+
+	// Query installed cc
+	queryInstalled(t, label, packageID, orgResMgmt)
+
+	// Approve cc
+	approveCC(t, packageID, orgResMgmt)
+
+	//sleep 5s for approve cache
+	time.Sleep(time.Duration(5) * time.Second)
+
+	// Query approve cc
+	queryApprovedCC(t, orgResMgmt)
+
+	// Check commit readiness
+	checkCCCommitReadiness(t, packageID, orgResMgmt)
+
+	// Commit cc
+	commitCC(t, orgResMgmt)
+
+	//sleep 5s for commit cache
+	time.Sleep(time.Duration(5) * time.Second)
```

Comment:
It would be preferable to use a strategy other than a simple Sleep. Is it possible to determine when ready?